workspaces/marketplace/plugins/marketplace-common/src/permissions.ts

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createPermission,
+  ResourcePermission,
+} from '@backstage/plugin-permission-common';
+
+/**
+ * @public
+ */
+export const RESOURCE_TYPE_EXTENSION_PLUGIN = 'extension-plugin';
+
+/**
+ * @public
+ */
+export const RESOURCE_TYPE_EXTENSION_PACKAGE = 'extension-package';
+
+/**
+ * @public
+ * Convenience type for extension plugin
+ */
+export type ExtensionPluginPermission = ResourcePermission<
+  typeof RESOURCE_TYPE_EXTENSION_PLUGIN
+>;
+
+/**
+ * @public
+ * Convenience type for extension package
+ */
+export type ExtensionPackagePermission = ResourcePermission<
+  typeof RESOURCE_TYPE_EXTENSION_PACKAGE
+>;
+
+/** This permission grants access to the endpoint that reads the configuration of the extension plugin
+ * @public
+ */
+export const extensionPluginReadPermission = createPermission({
```

Comment:
The variable name should include Configuration as well, to avoid confusing that this is a permission to read the plugin metadata.

IMHO, as long as we don't differentiate between Create update and Delete we should have two permissions:

* read
* write or update

If you want have separate for enableing and disabling the plugin I personally would use

* enable
* disable

Actually, that's then maybe not a configuration enable/disable, its then maybe `'extensions.plugin.enable`.

But I think we could start here also with `extensions.plugin.configuration.write`, because it just does this at the moment.